web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -352,6 +355,49 @@ export class BarChartComponent implements AfterViewInit {
     // Apply transition to all elements.
     brushBars.selectAll('rect').transition().duration(500);
 
+    // This function shows the tooltip and tags when the user hovers over a
+    // bar, or the empty area above it. It is declared locally to prevent the
+    // use of `this` in the callback function.
```

Comment:
I think the 'It is declared locally to prevent the use of `this` in the callback function.' is not quite accurate. If we declare it as a class method it will be more confusing. It seems that you do rely on `this` being set in a particular way to it would be worth mentioning that.

Reply:
Fixed!

Reply:
I looked a bit more into this issue: `this` works as expected in here because we use an arrow function. :-) [1] is an nice reference and [2] ans has some useful examples. This solution even has a name, it's called 'local fat arrow'. :-)

So I think we should say something along these lines:

> This function is used as a callback so use an arrow function to make `this` indicate the current object and not the context in which the callback is invoked.

[1] https://github.com/microsoft/TypeScript/wiki/'this'-in-TypeScript
[2] http://es6-features.org/#Lexicalthis